Assets/Scripts/SceneManager/LevelLoader.cs

```diff
@@ -0,0 +1,37 @@
+﻿using UnityEngine;
+using UnityEngine.SceneManagement;
+using DN.UI;
+using DN.SceneManagement.Data;
+
+namespace DN.SceneManagement
+{
+    /// <summary>
+    /// Here i do the actual loading of a scene through the buttons.
+    /// </summary>
+    public class LevelLoader : MonoBehaviour
+    {
+        [SerializeField]private LevelsData levelsData;
+
+        private HorizontalScroller horizontalScroller;
+
+        private int currentIndex = 0;
+        private int btnAmount;
+
+        private void Start()
+        {
+            horizontalScroller = GetComponent<HorizontalScroller>();
+
+            btnAmount = horizontalScroller.btn.Count + 1;
+        }
+
+        private void Update()
+        {
+            currentIndex = horizontalScroller.currentBtnIndex;
+        }
```

Comment:
Misschien handiger om de `currentIndex` te setten in HorizontalScroller, Nu ga je elke frame de currentIndex setten wat niet nodig is. 
Misschien de HorizontalScroller de LevelLoader in plaats van deze calss de HorizontalScroller.

Deze Class hoeft dan ook geen MonoBehaviour te zijn. dan kan je bij LevelLoader:
`public LevelLoader(int buttonCount)
{
btnAmount = buttonCount;
}`
En bij HorizontalScroller:
`LevelLoader levelLoader = new LevelLoader(btn.Count+1)`

Reply:
currentIndex zou gewoon moeten wijzen naar `horizontalScroller.currentBtnIndex`. Dit kan makkelijk met een Property :)